dask/highlevelgraph.py

```diff
@@ -843,8 +855,9 @@ def cull(self, keys: Iterable) -> "HighLevelGraph":
                 ret_layers[layer_name] = culled_layer
                 ret_key_deps.update(culled_deps)
 
+        ret_layers_keys = set(ret_layers.keys())
         ret_dependencies = {
-            layer_name: self.dependencies[layer_name] & ret_layers.keys()
+            layer_name: ret_layers_keys & self.dependencies[layer_name]
             for layer_name in ret_layers
```

Comment:
please add a comment like the above ones here too

Reply:
however I don't understand. In this case you're creating a new set anyway; it's not an in-place update. So it should not matter?

Reply:
It turns out swapping the order here didn't do anything, so I'll back that part out, but pre-computing `ret_layers_keys` is important.

To compute the intersection of two sets, you can iterate over one set and check if each element is in the other. If they're different sizes, it's faster to iterate over the smaller set and do checks against the larger one. When using the `&` operator, Python will ensure this if both arguments are really instances of `set` (I imagine `frozenset` would probably work too), but if they're only set-like (e.g. `dict_keys` or a custom `collections.abc.Set`) then it's not so smart and it's down to things like the lookup order for magic operator methods.

I'll add another comment to the code to explain it.

Reply:
Comment added.